cmd/cloudFoundryDeleteService.go

```diff
@@ -1,48 +1,108 @@
 package cmd
 
 import (
+	"bytes"
+	"fmt"
+	"strings"
+
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/telemetry"
 )
 
-func cloudFoundryDeleteService(CloudFoundryDeleteServiceOptions cloudFoundryDeleteServiceOptions, telemetryData *telemetry.CustomData) error {
+func cloudFoundryDeleteService(options cloudFoundryDeleteServiceOptions, telemetryData *telemetry.CustomData) error {
 
 	c := command.Command{}
 
 	// reroute command output to logging framework
 	c.Stdout(log.Entry().Writer())
 	c.Stderr(log.Entry().Writer())
 
-	cloudFoundryLogin(CloudFoundryDeleteServiceOptions, &c)
+	var err error
 
-	err := cloudFoundryDeleteServiceFunction(CloudFoundryDeleteServiceOptions.CfServiceInstance, &c)
-	if err != nil {
+	err = cloudFoundryLogin(options, &c)
+
+	if err == nil && options.CfDeleteServiceKeys == true {
+		err = cloudFoundryDeleteServiceKeys(options, &c)
+	}
+
+	if err == nil {
+		err = cloudFoundryDeleteServiceFunction(options.CfServiceInstance, &c)
+	}
+
+	if err == nil {
+		err = cloudFoundryLogout(&c)
+		if err != nil {
+			log.Entry().
+				WithError(err).
+				Fatal("Error while logging out occured.")
+		}
+	} else if err != nil {
 		cloudFoundryLogout(&c)
 		log.Entry().
 			WithError(err).
-			Fatal("Failed to delete Service!")
-		return err
+			Fatal("Error occured.")
 	}
 
-	cloudFoundryLogout(&c)
+	return err
```

Comment:
Just a suggestion: Since you ignore the error from logging out when there already is an error, you could simplify:
```
    logoutErr := cloudFoundryLogout(&c)
    if err == nil {
	err = logoutErr
    }
    if err != nil {
	log.Entry().WithError(err).Fatal("Failed to delete Service!")
    }
```

Reply:
Changed it so logging out error is not ignored anymore.